tests/Main.elm

```diff
@@ -240,6 +348,8 @@ errorTests =
         , test "wrong DOCTYPE keyword" (testDocumentError "<!DOCTYRP html><html></html>")
         , test "wrong DOCTYPE" (testDocumentError "<!DOCTYPE httl><html></html>")
         , test "wrong html tag" (testDocumentError "<!DOCTYPE html><document></document>")
+        , test "incomplete script1" (testDocumentError "<script>")
+        , test "incomplete script2 (PR#18 comment)" (testDocumentError "<script>'")
```

Comment:
@danneu Added tests for your cases.
Though I observed that without the change to `chompOneOrMore` this test still passes without infinite loop.
Nevertheless, accepting `champOneOrMore` does not harm the rest of the suites so I made the change.

Reply:
Thanks for looking into it. Infinite loops seem to be pretty easy for me to create with elm/parser so I'm not surprised that it's a me problem. :)